vmdb/spec/requests/api/vms_spec.rb

```diff
@@ -23,1008 +33,764 @@ def app
   end
 
   context "Vm accounts subcollection" do
-    it "query VM accounts subcollection with no related accounts" do
-      basic_authorize @cfme[:user], @cfme[:password]
-
-      vm = FactoryGirl.create(:vm_vmware)
-
-      @success = run_get "#{@cfme[:vms_url]}/#{vm.id}/accounts"
+    let(:acct1) { FactoryGirl.create(:account, :vm_or_template_id => vm.id, :name => "John") }
+    let(:acct2) { FactoryGirl.create(:account, :vm_or_template_id => vm.id, :name => "Jane") }
+    let(:vm_accounts_url)      { "#{vm_url(vm.id)}/accounts" }
+    let(:acct1_url)            { "#{vm_accounts_url}/#{acct1.id}" }
+    let(:acct2_url)            { "#{vm_accounts_url}/#{acct2.id}" }
+    let(:vm_accounts_url_list) { [acct1_url, acct2_url] }
+
+    context "query VM accounts subcollection with no related accounts" do
+      before do
+        api_basic_authorize
+
+        run_get vm_accounts_url
+      end
 
-      expect(@code).to eq(200)
-      expect(@result).to have_key("name")
-      expect(@result["name"]).to eq("accounts")
-      expect(@result["resources"]).to be_empty
+      include_examples "empty_query_result", :accounts
     end
 
-    it "query VM accounts subcollection with two related accounts" do
-      basic_authorize @cfme[:user], @cfme[:password]
-
-      vm = FactoryGirl.create(:vm_vmware)
-      acct1 = FactoryGirl.create(:account, :vm_or_template_id => vm.id, :name => "John")
-      acct2 = FactoryGirl.create(:account, :vm_or_template_id => vm.id, :name => "Jane")
+    context "query VM accounts subcollection with two related accounts" do
+      before do
+        api_basic_authorize
+        acct1
+        acct2
 
-      vm_accounts_url = "#{@cfme[:vms_url]}/#{vm.id}/accounts"
-      @success = run_get vm_accounts_url
+        run_get vm_accounts_url
+      end
 
-      expect(@success).to be_true
-      expect(@code).to eq(200)
-      expect(@result).to have_key("name")
-      expect(@result["name"]).to eq("accounts")
-      expect(@result["subcount"]).to eq(2)
-      expect(@result["resources"].size).to eq(2)
-      expect(resources_include_suffix?(@result["resources"], "href", "#{vm_accounts_url}/#{acct1.id}")).to be_true
-      expect(resources_include_suffix?(@result["resources"], "href", "#{vm_accounts_url}/#{acct2.id}")).to be_true
+      include_examples "query_result", :accounts, 2, :includes_hrefs => ["resources", :vm_accounts_url_list]
     end
 
-    it "query VM accounts subcollection with a valid Account Id" do
-      basic_authorize @cfme[:user], @cfme[:password]
-
-      vm = FactoryGirl.create(:vm_vmware)
-      acct1 = FactoryGirl.create(:account, :vm_or_template_id => vm.id, :name => "John")
+    context "query VM accounts subcollection with a valid Account Id" do
+      before do
+        api_basic_authorize
+        acct1
 
-      @success = run_get "#{@cfme[:vms_url]}/#{vm.id}/accounts/#{acct1.id}"
+        run_get acct1_url
```

Comment:
Is the `run_get` actually needed, or is the main side effect of this method to reference the object and create it?